src/netex_france/offer.rs

```diff
@@ -33,17 +36,56 @@ type JourneyPattern = VehicleJourney;
 pub struct OfferExporter<'a> {
     model: &'a Model,
     converter: Proj,
+    // Precalculation of the Stop Points per Route
+    route_points: BTreeMap<&'a str, Vec<Idx<StopPoint>>>,
+}
+
+fn calculate_route_points<'a>(model: &'a Model) -> BTreeMap<&'a str, Vec<Idx<StopPoint>>> {
+    model
+        .routes
+        .values()
+        .map(|route| {
+            // Unwrap is safe here since we are iterating on existing routes
+            let route_idx = model.routes.get_idx(&route.id).unwrap();
```

Comment:
```suggestion
        .iter()
        .map(|(route_idx, route)| {
```
Not better ?

Reply:
Much, much better. Thanks for this nice catch. Done in b1099530cfb087e12aa33bfad632afc13e68c760.